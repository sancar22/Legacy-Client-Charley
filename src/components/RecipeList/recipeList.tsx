import React from 'react';
import { IRecipe } from 'src/interfaces';
import Recipe from '../Recipe/recipe';

const RecipeList = ({
  recipeStore,
  viewAsSelf,
}: {
  recipeStore: IRecipe[];
  viewAsSelf: boolean;
}): JSX.Element => (
  <>
    {recipeStore.length
      && recipeStore.map((recipe: IRecipe) => (viewAsSelf ? (
          <Recipe
            key={recipe._id}
            recipe={recipe}
            remove={true}
            edit={true}
            save={false}
            self={true}
          />
      ) : (
          <Recipe
            key={recipe._id}
            recipe={recipe}
            remove={false}
            edit={false}
            save={true}
            self={false}
          />
      )))}
  </>
);

export default RecipeList;
