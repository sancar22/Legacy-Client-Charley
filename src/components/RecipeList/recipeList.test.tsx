import React from 'react';
import { create } from 'react-test-renderer';
import { Provider } from 'react-redux';
import reduxStore from '../../state/store';
import RecipeList from './recipeList';

const { store } = reduxStore();

describe('RecipeList component', () => {
  const recipeAdder = create(
    <Provider store={store}>
      <RecipeList recipeStore={[]}/>
    </Provider>,
  );
  test('match previous RecipeList snapshot', () => {
    expect(recipeAdder).toMatchSnapshot();
  });
});
