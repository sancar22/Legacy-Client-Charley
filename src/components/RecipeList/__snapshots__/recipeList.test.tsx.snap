// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeList component match previous RecipeList snapshot 1`] = `"0"`;
